[gatsby-remark-katex] Doesn’t work with gatsby-plugin-mdx

## Description

I followed [the instructions to install `gatsby-remark-katex`](https://www.gatsbyjs.org/packages/gatsby-remark-katex/) on a website using `gatsby-plugin-mdx` and it didn’t work. No errors, no nothing, it just didn’t transform the LaTeX in the Markdown.

### Steps to reproduce

In [this zip](https://github.com/gatsbyjs/gatsby/files/4270928/gatsby-remark-katex-with-gatsby-plugin-mdx.zip) there are two projects:

**`doesnt-work`:** This is a minimal project to show the issue. To create this project I just followed the instructions to install `gatsby-remark-katex`.

**`works`:** This is a workaround I found based on [this comment](https://github.com/gatsbyjs/gatsby/issues/6648#issuecomment-516722534) and [the documentation for `remark-math`](https://github.com/remarkjs/remark-math#packages), in which they suggest to use `rehype-katex` instead of `remark-html-katex`.

You can run the projects with the usual `npm install && npx gatsby develop` and visit `http://localhost:8000`.

In summary, the difference is instead of:

```
// doesnt-work
$ npm install gatsby-transformer-remark gatsby-remark-katex katex

// gatsby-config.js
{
  resolve: `gatsby-plugin-mdx`,
  options: {
    gatsbyRemarkPlugins: [`gatsby-remark-katex`]
  }
}
```

you do:

```
// works
$ npm install remark-math rehype-katex katex

// gatsby-config.js
{
  resolve: `gatsby-plugin-mdx`,
  options: {
    remarkPlugins: [require(`remark-math`)],
    rehypePlugins: [require(`rehype-katex`)]
  }
}
```

### Expected result

The LaTeX in the Markdown should be rendered.

### Actual result

The LaTeX in the Markdown appears unmodified.

### Environment

<details>

```
System:
  OS: macOS 10.15.3
  CPU: (4) x64 Intel(R) Core(TM) i5-7360U CPU @ 2.30GHz
  Shell: 5.7.1 - /bin/zsh
Binaries:
  Node: 13.7.0 - /usr/local/bin/node
  npm: 6.14.1 - /usr/local/bin/npm
Languages:
  Python: 2.7.16 - /usr/bin/python
Browsers:
  Chrome: 80.0.3987.122
  Safari: 13.0.5
```

</details>

What’s the best way for me to check?

I tried to clone `https://github.com/jlkiri/gatsby/`, switch to the `mdx-interop-default-exp` branch, go to `packages/gatsby-plugin-mdx/` directory`, run `npm install && npm link`, go to my `doesnt-work` example, and run `npm link gatsby-plugin-mdx`. But that failed with something to the effect of `Cannot find module 'webpack'`. I think it has to do with missing peer dependencies. There must be a better way to do this…

--- a/packages/gatsby-plugin-mdx/loaders/mdx-loader.js
+++ b/packages/gatsby-plugin-mdx/loaders/mdx-loader.js
@@ -4,6 +4,7 @@ const grayMatter = require(`gray-matter`)
 const unified = require(`unified`)
 const babel = require(`@babel/core`)
 const { createRequireFromPath, slash } = require(`gatsby-core-utils`)
+const { interopDefault } = require(`../utils/interop-default`)
 
 const {
   isImport,
@@ -158,6 +159,29 @@ ${contentWithoutFrontmatter}`
       return fileNode
     } else {
       return rawGetNode(id)
+    }
+  }
+
+  /**
+   * Support gatsby-remark parser plugins
+   */
+  for (let plugin of options.gatsbyRemarkPlugins) {
+    debug(`requiring`, plugin.resolve)
+    const requiredPlugin = interopDefault(require(plugin.resolve))
+    debug(`required`, plugin)
+    if (_.isFunction(requiredPlugin.setParserPlugins)) {
+      for (let parserPlugin of requiredPlugin.setParserPlugins(
+        plugin.options || {}
+      )) {
+        if (_.isArray(parserPlugin)) {
+          const [parser, parserPluginOptions] = parserPlugin
+          debug(`adding remarkPlugin with options`, plugin, parserPluginOptions)
+          options.remarkPlugins.push([parser, parserPluginOptions])
+        } else {
+          debug(`adding remarkPlugin`, plugin)
+          options.remarkPlugins.push(parserPlugin)
+        }
+      }
     }
   }
 

